packages/knex/src/index.ts

```diff
@@ -108,18 +108,23 @@ class DomainKnex {
       for (const key of domainSchema.keys()) {
         const column = decamelize(key);
         const value = domainSchema.values[key];
-        const type = value.type.constructor === Array ? value.type[0] : value.type;
+        const oneToMany = value.type.constructor === Array;
+        const type = oneToMany ? value.type[0] : value.type;
+
         if (type.isSchema) {
           const hostTableName = domainSchema.__.transient ? parentTableName : tableName;
-          const newPromise = this._createTables(hostTableName, type, seen);
+          const newPromise = this._createTables(oneToMany ? hostTableName : null, type, seen);
+          const foreignTable = column === 'parent' ? tableName : column;
```

Comment:
You should rather infer that it is a foreign key to the same table by checking that type is the same as the type of parent schema, not by checking field name